src/providers/ogr/qgsogrconnpool.h

```diff
@@ -61,7 +61,13 @@ class QgsOgrConnPoolGroup : public QObject, public QgsConnectionPoolGroup<QgsOgr
     Q_OBJECT
 
   public:
-    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ) { initTimer( this ); }
+    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ), mRefCount( 0 ) { initTimer( this ); }
+    void ref() { ++mRefCount; }
```

Comment:
Uhm? A `QgsOgrConnPoolGroup` is identified by the datasource path, so two providers for the same file will share the connection pool group.